libretroshare/src/serialiser/rstlvbase.cc

```diff
@@ -596,7 +596,7 @@ bool GetTlvString(const void *data, uint32_t size, uint32_t *offset,
 
     char *strdata = (char *) right_shift_void_pointer(tlvstart, TLV_HEADER_SIZE);
     uint32_t strsize = tlvsize - TLV_HEADER_SIZE; /* remove the header */
-    if (strsize <= 0) {
+    if (strsize == 0) {
```

Comment:
that test is wrong. We should test tlvsize<TLV_HEADER_SIZE. Otherwise if the result is < 0, then the uint32_t strsize will be large  and then the code will allocate some memory it shouldn't . It's probably a security issue!

Reply:
Done

Reply:
ReDone

Reply:
ReDone